### PR TITLE
Feature/salva service

### PR DESCRIPTION
**Pull request by danone771909**

### CODE REVIEW

File: battleship-logic/src/main/java/dreamteam/battleship/logic/arbiter/Arbiter.java [hunk 1]
```diff
@@ -27,4 +27,10 @@
      * won the game, because he destroyed all opponent's ships.
      */
     public abstract boolean isWinner();
+
+    /**
+     *
+     * @return number of ships that are alived
+     */
```
Comment (caroo7) [inline battleship-logic/src/main/java/dreamteam/battleship/logic/arbiter/Arbiter.java]: Maybe not necessary to write a comment when function name is clear and say what it does ;)

File: battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java [hunk 2]
```diff
@@ -83,8 +117,43 @@ private ShootResponse handleShoot(HttpSession session, int fieldNumber) {
...
+        logger.debug("Handling the shoot");
+        SalvaShootResponse response;
+        Map<Integer, Boolean> salvaShootingResponse = new HashMap<>();
+        Player winner = null;
+        MovementStatus ms = MovementStatus.SALVA_MODE;
+
+        for(int fieldNumber: fieldNumbers) {
+            MovementStatus status = controller.salvaShoot(fieldNumber, player);
+
+            switch(status) {
```
Comment (caroo7) [inline battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java]: Quite long method, maybe move switch - case to another one?

File: battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java [hunk 3]
```diff
@@ -83,8 +117,43 @@ private ShootResponse handleShoot(HttpSession session, int fieldNumber) {
         if(MovementStatus.WON.equals(status)){
```
Comment (caroo7) [inline battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java]: You can do this in one line using 3 arguments operator

File: battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java [hunk 4]
```diff
@@ -83,8 +117,43 @@ private ShootResponse handleShoot(HttpSession session, int fieldNumber) {
...
             response = winnerResponse();
         }else {
-            response = new ShootResponse(status, controller.getBoardForPlayer(player));
+            response = new ShootResponse(status);
```
Comment (caroo7) [inline battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/play/Playing.java]: You can do this in one line using 3 arguments operator

File: battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/preparation/GameController.java [hunk 5]
```diff
@@ -72,6 +72,19 @@ public MovementStatus shoot(int fieldNumber, Player player){
 
         return status;
     }
+    public MovementStatus salvaShoot(int fieldNumber, Player player){
+        MovementStatus status;
+
+        if(player.equals(player1)) {
```
Comment (caroo7) [inline battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/preparation/GameController.java]: Also here you can use 3 arguments operator

File: battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java [hunk 6]
```diff
@@ -52,4 +56,49 @@ public void testIfPlayerTermAreCorrect(){
...
+        when(organizer.myController()).thenReturn(controller);
+
+        MockHttpSession session = new MockHttpSession();
+        session.setAttribute("registration", registration);
+        session.setAttribute("placingShip", placingShip);
+        session.setAttribute("playerOrganizer", organizer);
+        playing.session = session;
+
+        try {
+            playing.afterPropertiesSet();
```
Comment (caroo7) [inline battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java]: Think about adding 'throws Exception' to method signature and let testNG to handle Exception.

File: battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java [hunk 7]
```diff
@@ -52,4 +56,49 @@ public void testIfPlayerTermAreCorrect(){
...
+        } catch(Exception e) {
+            assertTrue(false);
+        }
+        SalvaShootResponse status = playing.salvaShoot(shootingList);
+
+        System.out.println(status.status);
```
Comment (caroo7) [inline battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java]: It's probably not needed ;)

File: battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java [hunk 8]
```diff
@@ -52,4 +56,49 @@ public void testIfPlayerTermAreCorrect(){
...
+        return new Object[][] {
+                {new LinkedList<Integer>(Arrays.asList(1, 2, 3, 4, 5, 6, 7))},
+                {new LinkedList<Integer>(Arrays.asList(10, 2, 35, 42, 51, 68, 77))},
+                {new LinkedList<Integer>(Arrays.asList(11, 22, 33, 44, 55, 66, 73))},
+                {new LinkedList<Integer>(Arrays.asList(15, 12, 53, 84, 51, 62, 99))}
+        };
+    }
+    @Test(dataProvider = "shootingList")
+    public void testPlayerSalvaShooting(List<Integer> shootingList) {
+        Playing playing = new Playing();
```
Comment (caroo7) [inline battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java]: Maybe put this arrange instructions to setUp method. Test method is quite long

File: battleship-logic/src/main/java/dreamteam/battleship/logic/arbiter/Arbiter.java (hunk 1 above)
Comment (LIttleAncientForestKami) [inline battleship-logic/src/main/java/dreamteam/battleship/logic/arbiter/Arbiter.java]: ships that are alived - bad English

File: battleship-logic/src/main/java/dreamteam/battleship/logic/movement/DamageManager.java [hunk 9]
```diff
@@ -102,4 +102,8 @@ private void shipListCleanUp(Ship ship){
         }
     }
 
+    @Override
+    public int numberOfPlayerShoot() {
```
Comment (LIttleAncientForestKami) [inline battleship-logic/src/main/java/dreamteam/battleship/logic/movement/DamageManager.java]: Player Shoot? Bad English.

File: battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java (hunk 7 above)
Comment (egolesor) [inline battleship-service/src/test/java/dreamteam/battleship/service/springcontroller/play/PlayingTest.java]: Pull the master to your branch please, and resolve the conflicts (as you know the structure of the project is changed a bit), the response models must be in the directory model/response, without the postfix "response"

**Comment by LIttleAncientForestKami** [comment]
Just on English alone I'm not letting this in. :P

Also - what is salva? Please [check your English](http://dictionary.cambridge.org/dictionary/english/salvo) before pushing next time. 
 
:-1:

**Pull request closed without merging**